### DISCUSSION TITLE
How to do have the same endpoints for multiple methods ?

### DISCUSSION THREAD

**Discussion by JonathanCabezas** [RobinTail/express-zod-api#1975] [Q&A] [answered]
Hello,

I've read the whole documentation and searched through issues/discussions but I can't find an example of this.

The documentation says we can have the same endpoint handle multiple methods:

> 
> ### Multiple schemas for one route
> 
> Thanks to the DependsOnMethod class a route may have multiple Endpoints attached depending on different methods. It can also be the same Endpoint that handles multiple methods as well.
> 
> ```
> import { DependsOnMethod } from "express-zod-api";
> 
> // the route /v1/user has two Endpoints
> // which handle a couple of methods each
> const routing: Routing = {
>   v1: {
>     user: new DependsOnMethod({
>       get: yourEndpointA,
>       delete: yourEndpointA,
>       post: yourEndpointB,
>       patch: yourEndpointB,
>     }),
>   },
> };
> ```

However I have no idea how such an endpoint would look like, could you please provide examples for the get/delete and post/patch endpoint sharing above ? 

Thank you for your framework, I find it extremely well designed and helpful to structure my project !

**Comment by RobinTail** [comment, marked as answer]
> The documentation says we can have the same endpoint handle multiple methods:

yes, you can, @JonathanCabezas ,
all you need to make it possible is to set `methods` property with an array of ones when you do `factory.build()` (instead of `method`).

```ts
const yourEndpoint = yourFactory.build({
  methods: ["get", "delete"],
  /** other stuff and your implementation */
});
```

And then you can assign that endpoint both to `get` and `delete` prop of `DependsOnMethod::constructor` as in your original request. The handler of that endpoint will handle both methods of the same requested path. I hope it helps.

**Reply by JonathanCabezas** [reply]
First of all, thank you for your quick answer.

I think my question was unclear, as I already know what you answered.

My question is, how to you discriminate between the 2 method in your handler ?

**Reply by RobinTail** [reply]
The original idea of endpoint handling multiple methods is more suitable for the cases when the method does not matter.
But if it does, such an endpoint should use a middleware that returns `request.method`, which will come into `options` of that endpoint. See Middlewares in docs, @JonathanCabezas

**Reply by JonathanCabezas** [reply]
Thank you, that's understood !
